internal/cri/config/config.go

```diff
@@ -486,41 +453,17 @@ const (
 func ValidateImageConfig(ctx context.Context, c *ImageConfig) ([]deprecation.Warning, error) {
 	var warnings []deprecation.Warning
 
-	useConfigPath := c.Registry.ConfigPath != ""
-	if len(c.Registry.Mirrors) > 0 {
-		if useConfigPath {
-			return warnings, errors.New("`mirrors` cannot be set when `config_path` is provided")
-		}
-		warnings = append(warnings, deprecation.CRIRegistryMirrors)
-		log.G(ctx).Warning("`mirrors` is deprecated, please use `config_path` instead")
+	if len(c.Registry.MirrorsNoLongerSupported) > 0 {
+		return warnings, errors.New("`mirrors` is no longer supported since containerd v2.0, please use `config_path` instead")
 	}
 
-	if len(c.Registry.Configs) != 0 {
-		warnings = append(warnings, deprecation.CRIRegistryConfigs)
-		log.G(ctx).Warning("`configs` is deprecated, please use `config_path` instead")
+	if len(c.Registry.ConfigsNoLongerSupported) != 0 {
+		return warnings, errors.New("`configs` is no longer supported since containerd v2.0, please use `config_path` instead")
 	}
 
 	// Validation for deprecated auths options and mapping it to configs.
-	if len(c.Registry.Auths) != 0 {
-		if c.Registry.Configs == nil {
-			c.Registry.Configs = make(map[string]RegistryConfig)
-		}
-		for endpoint, auth := range c.Registry.Auths {
-			auth := auth
-			u, err := url.Parse(endpoint)
-			if err != nil {
-				return warnings, fmt.Errorf("failed to parse registry url %q from `registry.auths`: %w", endpoint, err)
-			}
-			if u.Scheme != "" {
-				// Do not include the scheme in the new registry config.
-				endpoint = u.Host
-			}
-			config := c.Registry.Configs[endpoint]
-			config.Auth = &auth
-			c.Registry.Configs[endpoint] = config
-		}
-		warnings = append(warnings, deprecation.CRIRegistryAuths)
```

Comment:
Can you remove `deprecation.CRIRegistryAuths`?

Reply:
@dmcgowan

Reply:
see https://github.com/containerd/containerd/pull/7526#issuecomment-1841925797